### PR TITLE
254 clinical review reviewed task

### PR DESCRIPTION
**Pull request by jackschofield23**
Implements #254 

### Description
Basic workflow validation on workflow API.
 - fixes the aide clinical review plugin
 - fixes conditional resolver to be to the workflow spec

### Status
**Ready**

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [x] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [x] User guide updated.
- [ ] I have updated the changelog
- [x] Any new files have copyright headers
- [x] Code coverage above 70%
- [x] TODO items resolved

### CODE REVIEW

File: src/ConditionsResolver/Parser/ConditionalParameterParser.cs [hunk 1]
```diff
@@ -243,29 +244,79 @@ private void ClearWorkflowParser()
...
+            if (subValues.Length > 3)
             {
-                var result = task.Metadata[metadataKey];
+                keyValue = subValues[3]?.Split('\'')[1];
+            }
+
+            var resultStr = null as string;
+            switch (subValueKey.ToLower())
+            {
+                case "task_id":
```
Comment (lillie-dae) [inline src/ConditionsResolver/Parser/ConditionalParameterParser.cs:268]: maybe we should implement a constants class for all these constant strings ?
could include patient details strings below?

File: src/ConditionsResolver/Parser/ConditionalParameterParser.cs [hunk 2]
```diff
@@ -243,29 +244,79 @@ private void ClearWorkflowParser()
...
+                    resultStr = task.TaskType;
+                    break;
+                case "previous_task_id":
+                    resultStr = task.PreviousTaskId;
+                    break;
+                case "error_msg":
+                    resultStr = task.Reason.ToString();
+                    break;
+                case "result":
+                    resultStr = GetValueFromDictionary(task.ResultMetadata, keyValue);
```
Comment (lillie-dae) [inline src/ConditionsResolver/Parser/ConditionalParameterParser.cs:290]: is this needed wouldn't TryGetValue work same here?

File: src/ConditionsResolver/Parser/ConditionalParameterParser.cs [hunk 3]
```diff
@@ -243,29 +244,79 @@ private void ClearWorkflowParser()
...
+                    break;
+                case "start_time":
+                    resultStr = task.TaskStartTime?.ToString(new CultureInfo("en-GB"));
```
Comment (lillie-dae) [inline src/ConditionsResolver/Parser/ConditionalParameterParser.cs:293]: is this the right CultureInfo to be having here?

**Review by lillie-dae — APPROVED** [review]

File: src/ConditionsResolver/Parser/ConditionalParameterParser.cs (hunk 3 above)
Comment (jackschofield23) [reply src/ConditionsResolver/Parser/ConditionalParameterParser.cs:293]: I'm not sure, if we do not include this it defaults to the american format when running in pipelines and can fail tests. We could potentially pull this out to config

File: src/ConditionsResolver/Parser/ConditionalParameterParser.cs (hunk 2 above)
Comment (jackschofield23) [reply src/ConditionsResolver/Parser/ConditionalParameterParser.cs:291]: I have used TryGetValue in that method, am keeping the method as it keeps this clean and provides nullchecks

File: src/ConditionsResolver/Parser/ConditionalParameterParser.cs (hunk 1 above)
Comment (jackschofield23) [reply src/ConditionsResolver/Parser/ConditionalParameterParser.cs]: Good idea, pulled them out to a constants file

File: src/ConditionsResolver/Parser/ConditionalParameterParser.cs (hunk 3 above)
Comment (jackschofield23) [reply src/ConditionsResolver/Parser/ConditionalParameterParser.cs]: removed for now to be consistant, we can change these later

**Pull request merged**